api/src/test/java/com/foro/api/TestTopicRepositoryIT.java

```diff
@@ -0,0 +1,149 @@
+
+@SpringBootTest
+public class TestTopicRepositoryIT extends AbstractTestNGSpringContextTests {
+
+    @Autowired
+    private TopicRepository topicRepository;
+
+    @Test
+    @Rollback(value = false)
+    @Transactional
+    void testCreateTopic() {
+
+        String title = "What means Spring";
+        String message = "I dont understand the difference between Spring and SpringBoot";
+        String creationDateString = "2021/05/04 11:30:00";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss");
+        LocalDateTime.parse("2023-11-07T17:24:36.859762975");
+        LocalDateTime creationDate = LocalDateTime.parse(creationDateString, formatter);
+        TopicStatus topicStatus = TopicStatus.CLOSED;
+        Integer idUser = 3;
+        Course course = Course.SPRINGBOOT;
+        Boolean deleteTopic = false;
+
+        TopicDTO topicInformation = TopicDTO.builder()
+                .title(title)
+                .message(message)
+                .creationDate(creationDate)
+                .topicStatus(topicStatus)
+                .idUser(idUser)
+                .course(course)
+                .deleteTopic(deleteTopic)
+                .build();
+
+        TopicDTO topic = topicRepository.save(topicInformation);
+        TopicDTO topicInformationInitial=new TopicDTO(topicInformation.getId(), idUser,title,message,creationDate,topicStatus,course,deleteTopic);
+
+
+        String titleModified = "What means SpringBoot";
+        String messageModified = "I dont understand the difference between Spring and SpringBoot";
+        String creationDateStringModified = "2021/05/04 11:30:00";
+        DateTimeFormatter formatterModified = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss");
+        LocalDateTime creationDateModified = LocalDateTime.parse(creationDateStringModified, formatterModified);
+        TopicStatus topicStatusModified = TopicStatus.CLOSED;
+        Integer idUserModified = 3;
+        Course courseModified = Course.SPRINGBOOT;
+        Boolean deleteTopicModified = false;
+
+        TopicDTO topicToBeModified = topicRepository.getReferenceById(topicInformation.getId());
+
+        topicToBeModified.setTitle(titleModified);
+        topicToBeModified.setMessage(messageModified);
+        topicToBeModified.setCreationDate(creationDateModified);
+        topicToBeModified.setTopicStatus(topicStatusModified);
+        topicToBeModified.setIdUser(idUserModified);
+        topicToBeModified.setCourse(courseModified);
+        topicToBeModified.setDeleteTopic(deleteTopicModified);
+
+        topicRepository.save(topicToBeModified);
+        Assert.assertEquals(topicToBeModified.getTitle(),"What means SpringBoot");
+        assertEqualsTopic(topicInformationInitial, topicToBeModified);
+
+        topicRepository.deleteById(topicToBeModified.getId());
+        boolean topicDeleted=topicRepository.findById(topicToBeModified.getId()).isPresent();
+        Assert.assertFalse(topicDeleted);
```

Comment:
Consider using `isEmpty()` and assert to true, to make more sense with the reading.
```java
        boolean topicDeleted=topicRepository.findById(topicToBeModified.getId()).isEmpty();
        Assert.assertTrue(topicDeleted);
```